### PR TITLE
20362 - Filing outputs

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#20362

*Description of changes:*

update filing outputs to work with alternate names

- Continuation out 
- AGM extension
- AGM location change
- Voluntary dissolution 

Some samples:

image

image

image

**This is for SP**:
**DEV link to compare**: https://dev.business.bcregistry.gov.bc.ca/FM1000050
image

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py [hunk 1]
```diff
@@ -121,6 +121,15 @@ def _get_receipt(business: any, filing: Filing, token):
...
     url = f'{current_app.config.get("PAYMENT_SVC_URL")}/{filing.storage.payment_token}/receipts'
+
+    business_number = ""
+    if business:
+        if business.is_alternate_name_entity and business.bn15:
+            business_number = business.bn15
+
+        if business.is_legal_entity and business.tax_id:
+            business_number = business.tax_id
+
```
Comment (ketaki-deodhar) [inline legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py:132]: To handle `tax_id` and `bn15` for both

File: legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py [hunk 2]
```diff
@@ -964,7 +964,7 @@ def test_update_ar_with_a_missing_filing_id_fails(session, client, jwt):
         filings = factory_completed_filing(legal_entity, ar)
 
         rv = client.put(
-            f"/api/v2/businesses/{identifier}/filings/{filings.id+1}",
+            f"/api/v2/businesses/{identifier}/filings/{filings.id + 1}",
```
Comment (ketaki-deodhar) [inline legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py:967]: to fix issues after `poetry run flake8 .`

File: legal-api/src/legal_api/services/business_details_version.py [hunk 3]
```diff
@@ -278,7 +278,7 @@ def get_office_revision(filing, business) -> dict:  # pylint: disable=too-many-l
 
         offices_current = (
             db.session.query(Office, null().label("changed"))
-            .filter(Office.change_filing_id == filing_id)
+            .filter(Office.change_filing_id <= filing_id)
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/business_details_version.py:281]: I am not sure about this. @argush3

File: legal-api/report-templates/dissolution.html [hunk 4]
```diff
@@ -27,7 +27,7 @@
             </tr>
          </table>
          <div class="business-name-header">
-                <label class="lbl-business-name">{{business.businessName}}</label>
+                <label class="lbl-business-name">{{business.legalName}}</label>
```
Comment (argush3) [inline legal-api/report-templates/dissolution.html:30]: I think this needs to be `businessName` as `legalName` is not the right value for firms.  The existing `businessName` value might not work properly as is right now though.

File: legal-api/src/legal_api/services/business_details_version.py [hunk 5]
```diff
@@ -812,6 +812,9 @@ def resolution_json(resolution_revision) -> dict:
     @staticmethod
     def business_revision_json(business_revision, business_json):
         """Return the business revision as a json object."""
+        if not business_revision:
+            return business_json
```
Comment (argush3) [inline legal-api/src/legal_api/services/business_details_version.py:816]: shouldn't need this.  Should never be null. 

I think it has to do with this [function](https://github.com/bcgov/lear/blob/1b40e6d07ba8b1d090b4ffd21ca3aaef102f1533/legal-api/src/legal_api/services/business_details_version.py#L222-L238) being incorrect.

File: legal-api/src/legal_api/services/business_details_version.py (hunk 3 above)
Comment (argush3) [reply legal-api/src/legal_api/services/business_details_version.py:281]: @vysakh-menon-aot let me know what issues you see with the updates Ketaki made when you have a minute.

I have attached an SP example below and although it is not directly applicable to the filings being tested, I think it still illustrates the retrieval of current and versioning data for offices.

Assuming the filing of interest was the most recent filing, I believe the updates Ketaki made here would be able to pick up the correct current and historical office entries.

image

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/business_details_version.py:281]: looks good

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py (hunk 2 above)
Comment (ketaki-deodhar) [reply legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py]: This is resolved after the latest update

**Review by argush3 — APPROVED** [review]

**Pull request merged**